lxd/instance/drivers/driver_qemu.go

```diff
@@ -7450,54 +7460,62 @@ func (d *qemu) getVsockID() (uint32, error) {
 	return uint32(vsockID), nil
 }
 
-// freeVsockID returns true if the given vsockID is not yet acquired.
-func (d *qemu) freeVsockID(vsockID uint32) bool {
-	c, err := lxdvsock.Dial(vsockID, shared.HTTPSDefaultPort)
-	if err != nil {
-		var unixErrno unix.Errno
+// acquireVsockID tries to occupy the given vsock Context ID.
+// If the ID is free it returns the corresponding file handle.
+func (d *qemu) acquireVsockID(vsockID uint32) (*os.File, error) {
+	vsockIDAcquistionLock.Lock()
+	defer vsockIDAcquistionLock.Unlock()
 
-		if !errors.As(err, &unixErrno) {
-			return false
-		}
+	f, err := os.OpenFile("/dev/vhost-vsock", syscall.O_RDWR, 0)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to open vhost socket: %w", err)
+	}
```

Comment:
We're going to want reverter in here to close the file handle on error.

Reply:
Yes, added. I have decided against using a returning cleanup hook since each invocation of this function (and upstream calls) either fail directly on error (which will trigger the revert in this function) or return the actual values until the point where the `vsockF` gets added to the `fdFiles` which is covered by the looped `_ = f.Close()`